// @aave/deploy-v3/dist/typechain/factories/contracts/EmptyContract__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmptyContract__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "get",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50607780601d6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c80636d4ce63c14602d575b600080fd5b600060405190815260200160405180910390f3fea2646970667358221220c2f8e63144d44ffa3d2d81f062b9c4e22b52da7d4f97f537ca0aba353600c4fd64736f6c634300080a0033";
const isSuperArgs = (xs) => xs.length > 1;
class EmptyContract__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.EmptyContract__factory = EmptyContract__factory;
EmptyContract__factory.bytecode = _bytecode;
EmptyContract__factory.abi = _abi;
