// uniswap-v2-deploy-plugin/dist/deployer/UniswapV2Deployer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniswapV2Deployer = void 0;
const ethers_1 = require("ethers");
const WETH9_json_1 = __importDefault(require("../util/WETH9.json"));
const artifacts = {
    UniswapV2Factory: require("@uniswap/v2-core/build/UniswapV2Factory.json"),
    UniswapV2Pair: require("@uniswap/v2-core/build/UniswapV2Pair.json"),
    UniswapV2ERC20: require("@uniswap/v2-core/build/UniswapV2ERC20.json"),
    UniswapV2Router02: require("@uniswap/v2-periphery/build/UniswapV2Router02.json"),
    WETH9: WETH9_json_1.default,
};
class UniswapV2Deployer {
    constructor(deployer) {
        this.deployer = deployer;
    }
    static async deploy(actor) {
        const deployer = new UniswapV2Deployer(actor);
        const weth9 = await deployer.deployWETH9();
        const factory = await deployer.deployFactory();
        const router = await deployer.deployRouter(factory.address, weth9.address);
        return {
            weth9,
            factory,
            router,
        };
    }
    async deployFactory() {
        return (await this.deployContract(artifacts.UniswapV2Factory.abi, artifacts.UniswapV2Factory.bytecode, [ethers_1.constants.AddressZero], this.deployer));
    }
    async deployWETH9() {
        return await this.deployContract(artifacts.WETH9.abi, artifacts.WETH9.bytecode, [], this.deployer);
    }
    async deployRouter(factoryAddress, weth9Address) {
        return (await this.deployContract(artifacts.UniswapV2Router02.abi, artifacts.UniswapV2Router02.bytecode, [factoryAddress, weth9Address], this.deployer));
    }
    async deployContract(abi, bytecode, deployParams, actor) {
        const factory = new ethers_1.ContractFactory(abi, bytecode, actor);
        return await factory.deploy(...deployParams);
    }
}
exports.UniswapV2Deployer = UniswapV2Deployer;
